static/app/components/multiPlatformPicker.tsx

```diff
@@ -75,13 +80,18 @@ function PlatformPicker(props: PlatformPickerProps) {
       (currentCategory?.platforms as undefined | string[])?.includes(platform.id);
 
     const popularTopOfAllCompare = (a: PlatformIntegration, b: PlatformIntegration) => {
-      // for the all category, put popular ones at the top
+      // for the all category, put popular ones at the top in the order they appear in the popular list
       if (category === 'all') {
+        if (isPopular(a) && isPopular(b)) {
+          // if both popular, maintain ordering from popular list
+          return popularIndex(a) - popularIndex(b);
```

Comment:
This would be a O(n^2 logn) sort right

Reply:
@Neo-Zhixing Yep, but n is only like 24 here so not so bad.